include/aws/iotdevice/secure_tunneling.h

```diff
@@ -73,4 +80,13 @@ struct aws_secure_tunnel *aws_secure_tunnel_new(const struct aws_secure_tunnelin
 AWS_SECURE_TUNNELING_API
 void aws_secure_tunnel_release(struct aws_secure_tunnel *secure_tunnel);
 
+AWS_SECURE_TUNNELING_API
+int aws_secure_tunnel_send_data(struct aws_secure_tunnel *secure_tunnel, const struct aws_byte_buf *data);
```

Comment:
Agreed, I was thinking about this last night, this should be a cursor, as it is read-only access. When the send complete callback is called, the memory the cursor references can be released, or you can copy it internally

Reply:
The reason it's a const byte_buf right now is because the serializer takes in a struct like this.
```
struct aws_iot_st_msg {
    enum aws_iot_st_message_type type;
    uint32_t streamId;
    int ignorable;
    struct aws_byte_buf payload;
};
```
The payload here I simply just point at the passed in byte_buf,  I could change this all to be cursor, but I would need to make some changes to the serializer as well.  Is that what you had in mind

Reply:
Now we pass in cursor :)